File: go/acronym/acronym.go
package acronym

import "strings"

// Abbreviate creates and returns an acronym for a string
func Abbreviate(s string) string {
	a := string(s[0])

	for i, char := range s {
		if char == ' ' || char == '-' {
			a += string(s[i+1])
		}
	}

	return strings.ToUpper(a)
}

File: csharp/acronym/Acronym.cs
using System;
using System.Linq;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        string[] separator = { " ", "-", "_" };
        string[] tokens = phrase.ToUpper().Split(separator, StringSplitOptions.RemoveEmptyEntries);
        return string.Join("", tokens.Select(token => token[0]));
    }
}